sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Exceptions.java

```diff
@@ -27,4 +27,19 @@ public static boolean isNameCacheStale(CosmosException e) {
         return isStatusCode(e, HttpConstants.StatusCodes.GONE)
                 && isSubStatusCode(e, HttpConstants.SubStatusCodes.NAME_CACHE_IS_STALE);
     }
+
+    public static boolean isThroughputControlRequestRateTooLargeException(CosmosException e) {
+        return isStatusCode(e, HttpConstants.StatusCodes.TOO_MANY_REQUESTS)
+            && isSubStatusCode(e, HttpConstants.SubStatusCodes.THROUGHPUT_CONTROL_REQUEST_RATE_TOO_LARGE);
+    }
+
+    public static boolean isPartitionCompletingSplittingException(CosmosException cosmosException) {
```

Comment:
isPartitionCompletingSplittingException -> isPartitionSplitEventException

Reply:
please don't rename. there is an exception with the same name. we should be consistent.

Reply:
will leave the method name as it is for now.

Reply:
Will keep it the same for now